src/units/attack_type.cpp

```diff
@@ -101,17 +101,19 @@ attack_type::attack_type(const config& cfg) :
 		else
 			icon_ = "attacks/blank-attack.png";
 	}
-	if(cfg.has_attribute("alignment") && (cfg["alignment"] == "neutral" || cfg["alignment"] == "lawful" || cfg["alignment"] == "chaotic" || cfg["alignment"] == "liminal")){
-		alignment_str_ = cfg["alignment"].str();
-	} else if(self_){
-		alignment_str_ =unit_alignments::get_string(self_->alignment());
+	//if alignment_ don't exist but self_ is here(like when called in lua), made alignment_ equals to unit alignment.
+	if(!alignment_ && self_){
```

Comment:
What is the point of considering `self_` here when it is accounted for in `attack_type::alignment()`?